### DISCUSSION TITLE
Vitest 2 and workspace coverage

### DISCUSSION THREAD

**Discussion by programmer24601** [vitest-dev/vitest#6095] [Q&A]
How is coverage supposed to work in vitest v2 in a workspace? I've got three vitest configs:
```
<root>/packages/frontend/vite.config.ts
<root>/packages/backend/vitest.config.ts
<root>/vitest.config.ts
```
In vitest v1.6.0 coverage exclude rules (defined as relative paths) in the "sub-package" configs were inherited in the root vitest config, so when running vitest run --coverage` from the root, all coverage rules were followed.

Now, this doesn't seem to be the case in vitest v2. Do I need to duplicate the coverage exclude rules at the root level or is there another way to "inherit" these rules from the "sub-package" configs?

Example:

`<root>/packages/backend/vitest.config.ts`:
```
import { defineProject } from 'vitest/config';

export default defineProject({
  test: {
    environment: 'node',
    globals: true,
    pool: 'forks',
    coverage: {
      reporter: ['text', 'json', 'html'],
      thresholds: {
        statements: 100,
        branches: 100,
        functions: 100,
        lines: 100
      },
      exclude: ['dist/**']
    }
  }
});
```

If `packages/backend/dist/**` is not explicitly excluded in the root package config, the code coverage (when run from the root directory) will report on it.

**Comment by AriPerkkio** [comment]
> Now, this doesn't seem to be the case in vitest v2. Do I need to duplicate the coverage exclude rules at the root level or is there another way to "inherit" these rules from the "sub-package" configs?

It works exactly as it did before. Note that in v2 we fixed [a bug where overriding defaults of `coverage.exclude` was not possible](https://github.com/vitest-dev/vitest/pull/5997). Now that you are overriding `coverage.exclude` with new value, the defaults are not applied. This is stated in documentation: https://vitest.dev/config/#coverage-exclude

> This option overrides all default options. Extend the default options when adding new patterns to ignore:
> 
> ```ts
> import { coverageConfigDefaults, defineConfig } from 'vitest/config'
> 
> export default defineConfig({
>   test: {
>     coverage: {
>       exclude: ['**/custom-pattern/**', ...coverageConfigDefaults.exclude]
>     },
>   },
> })
> ```

As `dist/**` is already part of default values, I would just recommend to remove whole `coverage.exclude`.

**Reply by programmer24601** [reply]
Dist in this case was just an example. What I'm trying to tell you is that upgrading from vitest v1.6 --> v2 has wrecked my coverage, flagging loads of things that were previously excluded.

It might me some quirk of the fix that means I need to reconfigure my exclusion rules. I'll raise a reproducible issue if I can't fix this.
